src/xwayland/xwm/mod.rs

```diff
@@ -1265,12 +1271,17 @@ impl X11Wm {
     }
 }
 
-fn handle_event<D: XwmHandler + 'static>(
+fn handle_event<D>(
     loop_handle: &LoopHandle<'_, D>,
     state: &mut D,
     xwm_id: XwmId,
     event: Event,
-) -> Result<(), ReplyOrIdError> {
+) -> Result<(), ReplyOrIdError>
+where
+    D: XwmHandler,
+    D: xwayland_shell::XWaylandShellHandler,
```

Comment:
If we require this here, we don't really support compositors using the old association method, do we?

Reply:
We require compositors to support the new association method. Whether they also handle the legacy method for old xwayland versions by checking for id-based associations in an idle callback (as I mention in the comment) is optional.